lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -546,25 +552,31 @@ def compile_method(symbol_name, constant, method)
             end
           end.join(', ')
 
-          parameters = "(#{parameters})" if parameters != ""
+          parameter_list = "(#{parameter_list})" if parameter_list != ""
+          signature_str = indented(compile_signature(signature, sanitized_parameters)) if signature
 
-          signature_str = indented(compile_signature(signature)) if signature
           [
             signature_str,
-            indented("def #{method_name}#{parameters}; end"),
+            indented("def #{method_name}#{parameter_list}; end"),
           ].compact.join("\n")
         end
 
         TYPE_PARAMETER_MATCHER = /T\.type_parameter\(:?([[:word:]]+)\)/
 
-        sig { params(signature: T.untyped).returns(String) }
-        def compile_signature(signature)
-          params = signature.arg_types
-          params += signature.kwarg_types.to_a
-          params << [signature.rest_name, signature.rest_type] if signature.has_rest
-          params << [signature.block_name, signature.block_type] if signature.block_name
+        sig { params(signature: T.untyped, parameters: T::Array[[Symbol, String]]).returns(String) }
+        def compile_signature(signature, parameters)
+          parameter_types = T.let(signature.arg_types, T::Array[[Symbol, T::Types::Base]])
```

Comment:
Why not directly composing a hash instead of an array of arrays that is then translated into a hash?

Reply:
Good call, done!